README.md

```diff
@@ -14,11 +14,279 @@ background](https://neilmitchell.blogspot.com/2020/01/one-haskell-ide-to-rule-th
+
+Available commands can be seen with:
+
+```bash
+stack ./install.hs help
+```
+
+Remember, this will take time to download a Stackage-LTS and an appropriate GHC for build
+haskell-language-server the first time.
+
+##### Install via cabal
+
+The install-script can be invoked via `cabal` instead of `stack` with the command
+
+```bash
+cabal v2-run ./install.hs --project-file install/shake.project <target>
+```
+
+or using the existing alias script
+
+```bash
+./cabal-hls-install <target>
+```
+
+Running the script with cabal on windows requires a cabal version greater or equal to `3.0.0.0`.
+
+For brevity, only the `stack`-based commands are presented in the following sections.
+
+##### Install specific GHC Version
+
+Install haskell-language-server for the latest available and supported GHC version (and hoogle docs):
+
+```bash
+stack ./install.hs hls
+```
+
+Install haskell-language-server for a specific GHC version (and hoogle docs):
+
+```bash
+stack ./install.hs hls-8.8.3
+stack ./install.hs data
+```
+
+The Haskell Language Server can also be built with `cabal v2-build` instead of `stack build`.
+This has the advantage that you can decide how the GHC versions have been installed.
+To see what GHC versions are available, the command `cabal-hls-install ghcs` can be used.
+It will list all *supported* GHC versions that are on the path for build with their respective installation directory.
+If you think, this list is incomplete, you can try to modify the PATH variable, such that the executables can be found.
+Note, that the targets `hls` and `data` depend on the found GHC versions.
+
+An example output is:
+
+```bash
+> cabal-hls-install ghcs
+******************************************************************
+Found the following GHC paths:
+ghc-8.6.5: /opt/bin/ghc-8.6.5
+ghc-8.8.3: /opt/bin/ghc-8.8.3
+
+******************************************************************
+```
+
+If your desired ghc has been found, you use it to install haskell-language-server.
+
+```bash
+cabal-hls-install hls-8.6.5
+cabal-hls-install data
+```
+
+## Project Configuration
+
+**For a full explanation of possible configurations, refer to [hie-bios/README](https://github.com/mpickering/hie-bios/blob/master/README.md).**
+
+haskell-language-server has some limited support via hie-bios to detect automatically
+your project configuration and set up the environment for GHC.
+The plan is to improve it to handle most use cases.
+
+However, for now, the more reliable way is using a `hie.yaml` file in the root
+of the workspace to **explicitly** describe how to setup the environment.
+For that you need to know what *components* have your project and the path
+associated with each one. So you will need some knowledge about
+[stack](https://docs.haskellstack.org/en/stable/build_command/#components) or [cabal](https://cabal.readthedocs.io/en/latest/cabal-commands.html?#cabal-v2-build) components.
+
```

Comment:
How would you feel about adding reference to using `gen-hie` here?

Reply:
Sure, good idea!

Reply:
@Avi-D-coder i've added it, feel free to rephrase it